README.md

```diff
@@ -157,43 +163,112 @@ Based on this information a user should be able to prepare their config with a l
+### Supported custom inference types and their examples
+### 1. vLLM
+#### Host your model
 ```
->> yoda generate --config config.yaml --concurrency --debug --inference
+podman run  -v ~/.cache/huggingface:/root/.cache/huggingface --env "HUGGING_FACE_HUB_TOKEN=<YOUR_TOKEN>" -p 8000:8000 --ipc=host  quay.io/vchalla/vllm-openai-cpu:latest --model llava-hf/llava-v1.6-mistral-7b-hf
 ```
-At present, we are using [openbmb/MiniCPM-Llama3-V-2_5](https://huggingface.co/openbmb/MiniCPM-Llama3-V-2_5) as our inference endpoint [hosted on an AI cluster](https://github.com/redhat-performance/yoda/tree/main/vqa-app#readme) to summarize the image. Here is how the output of updated panel data looks like after the inference.
-
-#### **Output**
+#### Yoda Usage
 ```
-The image is a bar chart with two bars representing different data points. The left bar is colored green and represents a value of 4.15, while the right bar is yellow and represents a value of 4.14. Both bars have additional information displayed as text: "41.65K req/s" for the green bar and "43.95K req/s" for the yellow bar. This suggests that the chart is comparing two quantities, possibly request rates, where the yellow bar has a slightly higher value than the green bar.
-``` 
-#### Note: `--deplot` and `--inference` falgs are mutually exclusive. 
-
-At the end `yoda generate` sub-command spits out a csv file called `panel_inference.csv` that a user can take a look at. 
+yoda generate --config ~/config.yaml --inference --inference-endpoint <YOUR_URL> --inference-api-key <YOUR_API_KEY> --inference-model llava-hf/llava-v1.6-mistral-7b-hf --inference-model-type vllm
+```
+**Note**: Please make sure that your vLLM hosted model responds through calls in the below format.
+```
+curl -X POST http://0.0.0.0:8000/v1/chat/completions \
+  -H "Content-Type: application/json" \
+  -d '{
+    "model": "llava-hf/llava-v1.6-mistral-7b-hf",
+    "messages": [
+      {
+        "role": "user",
+        "content": [
+          { "type": "text", "text": "What is happening in this image?" },
+          { "type": "image_url", "image_url": { "url": "data:image/png;base64,<YOUR_BASE64_ENCODED_IMAGE>" } }
+        ]
+      }
+    ]
+  }'
+```
+### 2. ollama
+#### Host your model
+```
+ollama pull llama3.2-vision
+```
+#### Yoda Usage
+```
+yoda generate --config ~/config.yaml --inference --inference-endpoint <YOUR_URL> --inference-api-key <YOUR_API_KEY> --inference-model llama3.2-vision --inference-model-type ollama
+```
+**Note**: Please make sure that your ollama hosted model responds through calls in the below format.
+```
+curl http://localhost:11434/api/generate   -d '{
+    "model": "llama3.2-vision",
+    "prompt": "Whats in the image?",
+    "stream": false,
+    "images": ["<YOUR_BASE64_ENCODED_IMAGE>"]
+  }'
+```
+Also we have two ollama hosted models in our grafana instance which are running on cpu. Please feel free to use them at your own risk (.i.e. very slow response times)
```

Comment:
since this is a public repo do we want to expose these URLs?

Reply:
Good point. Kept them with the intention that they can only be accessed by VPN. But removed their references redirecting to performance and scale infrastructure document.